crypto/crmf/crmf_lib.c

```diff
@@ -720,6 +720,7 @@ X509 *OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert(OSSL_CRMF_ENCRYPTEDVALUE *ecert,
     EVP_CIPHER_CTX_set_padding(evp_ctx, 0);
 
     if (!EVP_DecryptInit(evp_ctx, cipher, ek, iv)
+            || eksize != (size_t)EVP_CIPHER_key_length(cipher)
```

Comment:
you need to check the eksize immediatly **after** the EVP_PKEY_decrypt.
best practice would be to do that with constant time code.
and to call ERR_clear_error before looking at the result code.

Reply:
Hopefully I got it now.

Reply:
It meanwhile turned out the constant-time style of checking the decryption result is not needed in the given context - see https://github.com/openssl/openssl/issues/17319#issuecomment-998863323.